src/app/components/generic-modal/generic-modal.component.ts

```diff
@@ -13,19 +15,29 @@ export class GenericModalComponent implements OnInit {
   formData: any;
   addDataFunction: (formData: any) => void;
   restaurants: IRestaurant[] = [];
+  dishes: IDish[] = [];
+  chefs: IChef[] = [];
 
   constructor(
     public dialogRef: MatDialogRef<GenericModalComponent>,
     @Inject(MAT_DIALOG_DATA) public data: any,
-    private restaurantService: RestaurantsService
+    private restaurantService: RestaurantsService,
+    private dishService: DishService,
+    private chefService: ChefService,
   ) {
     this.formData = data && data.formFields ? { ...data } : null;
     this.addDataFunction = data && data.addDataFunction ? data.addDataFunction : null;
   }
 
   ngOnInit(): void {
-    this.restaurantService.getRestaurants().subscribe((data) => {
-      this.restaurants = data;
+    this.restaurantService.getRestaurants().subscribe((restaurants) => {
+      this.restaurants = restaurants;
+    });
+    this.dishService.getDishes().subscribe((dishes) => {
+      this.dishes = dishes;
+    });
+    this.chefService.getChefs().subscribe((chefs) => {
+      this.chefs = chefs;
```

Comment:
A generic modal shouldn't know his objects, if we used this modal in a very big project it would become inefficient.
Try using props for this